### PR TITLE
add all discovered nodes to the updater

### PR DESCRIPTION
**Pull request by aplanas**
Seems that the `onadmin_cloudupgrade_clients` is not adding the clients into the `elements` deployment attribute

### CODE REVIEW

**Comment by jdsn** [comment]
This can already be run as `mkcloud [...] onadmin+cloudupgrade_clients [...]`
Do we really need this as a new step?

File: scripts/mkcloud [hunk 1]
```diff
@@ -1221,6 +1222,12 @@ function runupdate
     return $?
 }
 
+function runupdate_clients
```
Comment (jdsn) [inline scripts/mkcloud]: declaring this as a function is not enough to make it a step, it is not yet known by the step validator

**Comment by aplanas** [comment]
@jdsn the `onadmin+` seem to work. Thanks!

I will rebase it to maintain the fix from the first commit

**Comment by jdsn** [comment]
We verified this function that you are changing here at SAP a few days ago, and it worked for them. So I am wondering why this change is needed.

**Comment by aplanas** [comment]
@jdsn In my scenario it just deployed the barclamp and no single node was updated: elements.updater was empty. Is easy to replicate the issue:

https://ci.suse.de/view/Cloud/view/Cloud7%20Exotic/job/cloud-mkcloud7-job-simpletest-s390x/configure

**Comment by bmwiedemann** [comment]
@dirkmueller could also hint at a bug in the product - should it matter if nodes are brought in via autoyast or crowbar_register? or is it because of s390?

**Comment by aplanas** [comment]
@bmwiedemann we are using setuplonelynodes and crowbar_register in s390x. From there is the same that a normal setup.

But is expected that updater barclamp will automatically add all the nodes? For me was a surprise that in the code there is no node selection, because an updater that add all the nodes seems a bit dangerous to me.

In any case I will check the updater barclamp in case that we are hiding something

**Comment by dirkmueller** [comment]
If it is leaving out crowbar_register'ed nodes then that's definitely a product defect and shouldn't be hacked around in automation

**Comment by bmwiedemann** [comment]
I cherry-picked the typo-fix to master.

**Comment by aplanas** [comment]
Well, if the fix must be in other place, lets close this!

@bmwiedemann thanks for the cherry-pick!

**Pull request closed without merging**